keras_rs/src/utils/loss_and_metric_utils.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional
+
+from keras import ops
+
+from keras_rs.src import types
+from keras_rs.src.utils.keras_utils import check_rank
+from keras_rs.src.utils.keras_utils import check_shapes_compatible
+
+
+def process_inputs(
```

Comment:
This have a more specific name, maybe something like `standardize_ranks` or `check_ranks_and_shapes`.